tests/MongoDB.Driver.Tests/AggregateGraphLookupEnumerableFromOrToTests.cs

```diff
@@ -0,0 +1,241 @@
+                },
+                new ManyToOneResult
+                {
+                    Id = 2,
+                    From = new[] { 3, 4 },
+                    To = 2,
+                    Matches = new List<ManyToOne>()
+                }
+            };
+            var collection = database.GetCollection<ManyToOne>(collectionName);
+
+            var result = collection
+                .Aggregate()
+                .GraphLookup(
+                    from: collection,
+                    connectFromField: x => x.From,
+                    connectToField: x => x.To,
+                    startWith: x => x.From,
+                    @as: (ManyToOneResult x) => x.Matches)
+                .ToList();
+
+            result.Count.Should().Be(2);
+            result[0].ToBsonDocument().Should().Be(expectedResult[0].ToBsonDocument());
+            result[1].ToBsonDocument().Should().Be(expectedResult[1].ToBsonDocument());
+        }
+
+        [SkippableFact]
+        public void GraphLookup_with_one_to_many_parameters_should_return_expected_result()
+        {
+            RequireServer.Check().Supports(Feature.AggregateGraphLookupStage);
+            var database = GetDatabase();
+            var collectionName = "collectionOneToMany";
+            EnsureTestDataOneToMany(database, collectionName);
+            var expectedResult = new OneToManyResult[]
+            {
+                new OneToManyResult
+                {
+                    Id = 1,
+                    From = 1,
+                    To = new[] { 2, 3 },
+                    Matches = new List<OneToMany>()
+                },
+                new OneToManyResult
+                {
+                    Id = 2,
+                    From = 2,
+                    To = new[] { 3, 4 },
+                    Matches = new List<OneToMany> { new OneToMany { Id = 1, From = 1, To = new[] { 2, 3 } } }
+                }
+            };
+            var collection = database.GetCollection<OneToMany>(collectionName);
+
+            var result = collection
+                .Aggregate()
+                .GraphLookup(
+                    from: collection,
+                    connectFromField: x => x.From,
+                    connectToField: x => x.To,
+                    startWith: x => x.From,
+                    @as: (OneToManyResult x) => x.Matches)
+                .ToList();
+
+            result.Count.Should().Be(2);
+            result[0].ToBsonDocument().Should().Be(expectedResult[0].ToBsonDocument());
+            result[1].ToBsonDocument().Should().Be(expectedResult[1].ToBsonDocument());
+        }
+
+        [SkippableFact]
+        public void GraphLookup_with_one_to_one_parameters_should_return_expected_result()
+        {
+            RequireServer.Check().Supports(Feature.AggregateGraphLookupStage);
+            var database = GetDatabase();
+            var collectionName = "collectionOneToOne";
+            EnsureTestDataOneToOne(database, collectionName);
```

Comment:
Ditto above

Reply:
I feel like separate private methods are quite readable, so I leave it as is.